engine/src/flutter/lib/web_ui/lib/channel_buffers.dart

```diff
@@ -131,10 +131,17 @@ class ChannelBuffers {
 
   final Map<String, _Channel> _channels = <String, _Channel>{};
 
+  @visibleForTesting
+  bool debugPrintOverflowWarning = true;
+
   void push(String name, ByteData? data, PlatformMessageResponseCallback callback) {
     final _Channel channel = _channels.putIfAbsent(name, () => _Channel());
     if (channel.push(_StoredMessage(data, callback))) {
       assert(() {
+        if (!debugPrintOverflowWarning) {
```

Comment:
This is better than nothing, but I think it might be better if instead of having a special boolean flag that is exposed as `@visibleForTesting` that we actually just added a platform message listener instead, even if it does nothing and just silences these log messages. Is that simple enough to do?

Reply:
That's what I tried first, but it didn't work for a reason that I don't remember now (I wrote this PR a few weeks ago). I'll try it again and document why adding a listener didn't work.

Reply:
I switched this to `printWarning` too (which seems to not show anything in the test output).